src/Template/Provider/BaseProvider.php

```diff
@@ -140,12 +140,31 @@ public function installTemplateSource(Template\TemplateSource $templateSource):
             throw Exception\InvalidTemplateSourceException::forFailedInstallation($templateSource);
         }
 
-        $this->messenger->done();
-        $this->messenger->newLine();
-
         // Make sure installed sources are handled by Composer's class loader
         $loader = Resource\Local\Composer::createClassLoader(dirname($composerJson));
         $loader->register(true);
+
+        // Look up installed package
+        $composer = Resource\Local\Composer::createComposer(dirname($composerJson));
+        $repository = $composer->getRepositoryManager()->getLocalRepository();
+        $installedPackage = $repository->findPackage($package->getName(), new Semver\Constraint\MatchAllConstraint());
+
+        // Overwrite package from template source with actually installed template
+        if (null !== $installedPackage) {
+            $templateSource->setPackage($installedPackage);
+        }
+
+        // Show installed template version
+        $this->messenger->clearLine();
+        $this->messenger->progress(
+            sprintf(
+                'Installing project template (<info>%s</info>)...',
+                $templateSource->getPackage()->getPrettyVersion(),
+            ),
+            ComposerIO\IOInterface::NORMAL,
+        );
```

Comment:
Printing out the progress with the semver of the installable package works perfectly well. 👍 

**TL;DR;**
1. Let's also provide the template version when skipping the custom version selection so users actually see which version is installed. (Especially when they choose the latest stable.)
2. However, wouldn't the snippet than be redundant?

---

**Let me explain**

1. What makes this print-out (specific for selected versions) so special in comparison to the progess print in line 124 for latest stable?

```php
# line 124-132

$this->messenger->progress(
            sprintf(
                'Installing project template%s...',
                $templateSource->shouldUseDynamicVersionConstraint()
                    ? ''
                    : sprintf(' (<info>%s</info>)', $templateSource->getPackage()->getPrettyVersion()),
            ),
            ComposerIO\IOInterface::NORMAL,
        );
```

2. I would argue that it makes huge sense to display the tried version for latest stable as well:

```php
$this->messenger->progress(
            sprintf(
                'Installing project template (<info>%s</info>)...',
                $templateSource->getPackage()->getPrettyVersion(),
            ),
            ComposerIO\IOInterface::NORMAL,
        );
```

I have not yet committed the change, though, because I fear that there's something up the separation of both `$this->messenger->progress()` or else the newly added code would be redundant.

Reply:
Addendum: Since we're display the progress here, we're replacing that line with the actually installed version despite what package was initially found. All in all it would then of course make sense to keep your proposition as is. Thanks for the walk-through. 💪